tornjak-frontend/src/components/entry-create.tsx

```diff
@@ -132,122 +203,172 @@ class CreateEntry extends Component {
+          this.setState({
+            selectorsList: this.props.globalSelectorInfo["aws_iid"]
+          });
+        }
+        else if (serverNodeAtt === "gcp_iit") {
+          this.setState({
+            selectorsList: this.props.globalSelectorInfo["gcp_iit"]
+          });
+        }
+        else if (serverNodeAtt === "k8s_sat") {
+          this.setState({
+            selectorsList: this.props.globalSelectorInfo["k8s_sat"]
+          });
+        }
+        else if (serverNodeAtt === "k8s_psat") {
+          this.setState({
+            selectorsList: this.props.globalSelectorInfo["k8s_psat"]
+          });
+        }
+      } else if (parentId !== "") {
+        let agentId = parentId;
+        // Check if parent ID is not canonical ID, best effort try to match it to an Agent ID for selectors
+        if (!this.props.globalAgentsList.map(e => this.SpiffeHelper.getAgentSpiffeid(e)).includes(parentId)) {
+          let fEntries = this.props.globalEntriesList.filter(e => this.SpiffeHelper.getEntrySpiffeid(e) === parentId);
           if (fEntries.length > 0) {
             let entry = fEntries[0];
             let canonicalAgentId = this.SpiffeHelper.getCanonicalAgentSpiffeid(entry, this.props.globalAgentsList)
             if (canonicalAgentId !== "") {
               agentId = canonicalAgentId;
             }
           }
-      }
+        }
 
-      for (let i = 0; i < globalAgentsWorkLoadAttestorInfo.length; i++) {
-        if (agentId === globalAgentsWorkLoadAttestorInfo[i].spiffeid) {
+        for (let i = 0; i < globalAgentsWorkLoadAttestorInfo.length; i++) {
+          if (agentId === globalAgentsWorkLoadAttestorInfo[i].spiffeid) {
+            let assignedPlugin = globalAgentsWorkLoadAttestorInfo[i].plugin;
+            if (assignedPlugin === "Docker") {
+              this.setState({
+                selectorsList: this.props.globalWorkloadSelectorInfo["Docker"]
+              });
+            } else if (assignedPlugin === "Kubernetes") {
+              this.setState({
+                selectorsList: this.props.globalWorkloadSelectorInfo["Kubernetes"]
+              });
+            } else if (assignedPlugin === "Unix") {
+              this.setState({
+                selectorsList: this.props.globalWorkloadSelectorInfo["Unix"]
+              });
+            }
+            agentSelectorSet = true;
+          }
+        }
+        if (!agentSelectorSet) {
           this.setState({
-            selectorsList: this.props.globalWorkloadSelectorInfo[globalAgentsWorkLoadAttestorInfo[i].plugin]
+            selectorsList: [],
+            selectorsListDisplay: "Select Selectors",
           });
-          agentSelectorSet = true;
         }
       }
-      if (!agentSelectorSet) {
-        this.setState({
-          selectorsList: [],
-          selectorsListDisplay: "Select Selectors",
-        });
-      }
     }
   }
 
-  onChangeTtl(e) {
+  // e - any for now will be explicitly typed on currently open entry create PR
```

Comment:
add TODO as above (for each instance), i would also create an issue to do a sweep through of this and state what we are looking for just in case it gets lost in the future, or context is missing.

Reply:
Done!